server/middleware/setUpCsrf.ts

```diff
@@ -0,0 +1,32 @@
+import { Router } from 'express'
+import { csrfSync } from 'csrf-sync'
+
+const testMode = process.env.NODE_ENV === 'test'
+
+export default function setUpCsrf(): Router {
+  const router = Router({ mergeParams: true })
+
+  // CSRF protection
+  if (!testMode) {
+    const {
+      csrfSynchronisedProtection, // This is the default CSRF protection middleware.
+    } = csrfSync({
+      // By default, csrf-sync uses x-csrf-token header, but we use the token in forms and send it in the request body, so change getTokenFromRequest so it grabs from there
+      getTokenFromRequest: req => {
+        // eslint-disable-next-line no-underscore-dangle
+        return req.body._csrf || req.query._csrf
```

Comment:
this is slightly different to other impls as multipart post sends this via query parameters rather than just the body.

The old csurf library checked body, query and then a range of headers for it.